### PR TITLE
[fix](inverted index) Fix undefined behavior in write_v2

### PR DESCRIPTION
**Pull request by zzzxl1993**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:
1. Fix undefined behavior in finalize_output_dir caused by exception testing

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by zzzxl1993** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/inverted_index_file_writer.cpp [hunk 1]
```diff
@@ -268,38 +270,39 @@ Status InvertedIndexFileWriter::write_v1() {
...
+            eptr = std::current_exception();
             auto index_path = InvertedIndexDescriptor::get_index_file_path_v1(
                     _index_path_prefix, index_id, index_suffix);
-            LOG(ERROR) << "CLuceneError occur when write_v1 idx file " << index_path
-                       << " error msg: " << err.what();
-
-            return Status::Error<ErrorCode::INVERTED_INDEX_CLUCENE_ERROR>(
-                    "CLuceneError occur when write_v1 idx file: {}, error msg: {}", index_path,
-                    err.what());
+            err_msg = "CLuceneError occur when write_v1 idx file " + index_path +
```
Comment (csun5285) [inline be/src/olap/rowset/segment_v2/inverted_index_file_writer.cpp:281]: 这得 return Error, 写错一个不能继续写了

File: be/src/olap/rowset/segment_v2/inverted_index_file_writer.cpp [hunk 2]
```diff
@@ -268,38 +270,39 @@ Status InvertedIndexFileWriter::write_v1() {
...
+                      " error msg: " + err.what();
         }
     }
+
+    // Close and clean up
+    finalize_output_dir(out_dir);
```
Comment (csun5285) [inline be/src/olap/rowset/segment_v2/inverted_index_file_writer.cpp:287]: 在 finalize_output_dir 里面把 out_dir 置空就行

Comment (zzzxl1993) [reply be/src/olap/rowset/segment_v2/inverted_index_file_writer.cpp:287]: _CLDECDELETE会置空，finalize_output_dir参数是值拷贝，里面的指针置空并不能影响外面

File: be/src/olap/rowset/segment_v2/inverted_index_file_writer.cpp (hunk 1 above)
Comment (zzzxl1993) [reply be/src/olap/rowset/segment_v2/inverted_index_file_writer.cpp:281]: done

**Comment by zzzxl1993** [comment]
run buildall

**Review by csun5285 — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

File: be/src/olap/rowset/segment_v2/inverted_index_file_writer.cpp (hunk 2 above)
Comment (airborne12) [reply be/src/olap/rowset/segment_v2/inverted_index_file_writer.cpp]: why not pass the reference of the pointer？

**Pull request merged**